### ISSUE TITLE
Regarding the async iterator

### ISSUE THREAD

**Issue by isopen** [webpack/webpack#7298]
# Bug report

```
async function test123() {
  for await (const i of [1, 2, 3]) {
    console.log(123);
  }
}
test123()
```
https://github.com/isopen/soc/tree/master/frontend

**What is the current behavior?**
syntax error

**If the current behavior is a bug, please provide the steps to reproduce.**
npm install
Paste this piece of code into the center of my application.
npm start

**What is the expected behavior?**
should work the same as in `ng serve`

**Other relevant information:**
webpack version:  latest
Node.js version: 8.9.4
Operating System: debian 9.3

**Comment by evilebottnawi** [comment]
@isopen acorn can't parse this 

https://astexplorer.net/#/gist/c35d1e4a22a671c17dcaa88c2feb818a/d81c6eb1024e0b490a4d674ce5afa101d11c85f2

**Comment by isopen** [comment]
This is some kind of magic in my example that's earned today via webpack. Yesterday did not work.
About the design itself, it's a relatively new feature :)

**Comment by ooflorent** [comment]
This should be fixed since it reached stage 4.

**Comment by isopen** [comment]
He worked in my project yesterday :)
The day before yesterday there was an error in the line https://astexplorer.net/#/gist/c35d1e4a22a671c17dcaa88c2feb818a/d81c6eb1024e0b490a4d674ce5afa101d11c85f2

**Comment by ooflorent** [comment]
I cannot reproduce. This has been fixed when acorn was updated to version 5.4.0.
